lib/smart_proxy_dhcp_infoblox/dhcp_infoblox_plugin.rb

```diff
@@ -14,9 +15,10 @@ class Plugin < ::Proxy::Provider
     requires :dhcp, '>= 1.13'
 
     load_classes ::Proxy::DHCP::Infoblox::PluginConfiguration
-    load_validators :record_type_validator => ::Proxy::DHCP::Infoblox::RecordTypeValidator
+    load_validators :record_type_validator => ::Proxy::DHCP::Infoblox::RecordTypeValidator, :used_ips_search_type_validator => ::Proxy::DHCP::Infoblox::UsedIpsSearchTypeValidator
     load_dependency_injection_wirings ::Proxy::DHCP::Infoblox::PluginConfiguration
 
     validate :record_type, :record_type_validator => true
+    validate :used_ips_search_type, :used_ips_search_type_validator => true
```

Comment:
Instead of creating a custom validator, you can also use the Enum validator (introduced in https://github.com/theforeman/smart-proxy/commit/685c026e30ade098f1748fb8c5a0dd16e993efec)
```suggestion
    validate :used_ips_search_type, enum: %w[record_type used]
```

If you do so, please raise the minimum version on line 14/15 (`requires :dhcp`) to `>= 3.2`.